packages/relay/src/lib/hbarlimiter/index.ts

```diff
@@ -108,44 +110,60 @@ export default class HbarLimit {
     if (this.remainingBudget <= 0) {
       this.hbarLimitCounter.labels(mode, methodName).inc(1);
       this.logger.warn(
-        `${requestIdPrefix} HBAR rate limit incoming call: remainingBudget=${this.remainingBudget}, total=${this.total}, resetTimestamp=${this.reset}`,
+        `${requestIdPrefix} HBAR rate limit incoming call: remainingBudget=${this.remainingBudget}, total=${this.total}, resetTimestamp=${this.reset}.`,
       );
       return true;
+    } else {
+      this.logger.trace(
+        `${requestIdPrefix} HBAR rate limit not reached: remainingBudget=${this.remainingBudget}, total=${this.total}, resetTimestamp=${this.reset}.`,
+      );
+      return false;
     }
-
-    this.logger.trace(
-      `${requestIdPrefix} HBAR rate limit not reached. ${this.remainingBudget} out of ${this.total} tℏ left in relay budget until ${this.reset}.`,
-    );
-
-    return false;
   }
 
   /**
-   * Determines whether a preemptive HBAR rate limit should be applied based on the remaining budget and the transaction fee.
+   * Preemptively limits HBAR transactions based on the estimated total fee for file transactions and the remaining budget.
+   * This method checks if the caller is whitelisted and bypasses the limit if they are. If not, it calculates the
+   * estimated transaction fees based on the call data size and file append chunk size, and throws an error if the
+   * remaining budget is insufficient to cover the estimated fees.
    *
-   * Bypass if the originalCallerAddress is whitelisted
-   *
-   * @param {string} originalCallerAddress - The address of the original caller making the request.
-   * @param {number} transactionFee - The transaction fee in tinybars to be checked against the remaining budget.
-   * @param {string} [requestId] - An optional unique request ID for tracking the request.
-   * @returns {boolean} - Returns `true` if the rate limit should be preemptively enforced, otherwise `false`.
+   * @param {string} originalCallerAddress - The address of the caller initiating the transaction.
+   * @param {number} callDataSize - The size of the call data that will be used in the transaction.
+   * @param {number} fileChunkSize - The chunk size used for file append transactions.
+   * @param {string} requestId - The request ID for tracing the request flow.
+   * @throws {JsonRpcError} Throws an error if the total estimated transaction fee exceeds the remaining HBAR budget.
    */
-  shouldPreemtivelyLimit(originalCallerAddress: string, transactionFee: number, requestId?: string): boolean {
-    if (!this.enabled) {
-      return false;
-    }
-
+  shouldPreemtivelyLimitFileTransactions(
```

Comment:
Please, keep the old logic which returns booleans instead of throwing errors from inside this method.

Reply:
yeah good point! Pushed in the updates reverting the old logic.